src/binary-reader-linker.cc

```diff
@@ -163,6 +163,11 @@ Result BinaryReaderLinker::OnImportGlobal(uint32_t import_index,
   return Result::Ok;
 }
 
+Result BinaryReaderLinker::OnFunctionCount(uint32_t count) {
+  binary->debug_names.resize(count + binary->function_imports.size());
```

Comment:
The "count" is the count of function indices+names to follow, not the maximum function index. In addition, function imports aren't counted separately.

IOW, it's legal for count = 1 to be followed by index = 1000 and a name, which would blow up below.